src/java/servlets/addTag.java

```diff
@@ -39,28 +39,26 @@
     protected void processRequest(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
         response.setContentType("text/html;charset=UTF-8");
+        int position;
         try (PrintWriter out = response.getWriter()) {
             if (request.getParameter("UID") != null) {
-                        int ctr = 1;
-                        int UID=parseInt(request.getParameter("UID"));
-                        while (new TagButton(UID, ctr).exists()) {
-                            ctr++;
-                        }
-                        new TagButton(UID, ctr, "new", "description");
-                        out.print(""+ctr);
-                        return;
-                    }
+                //FIXME if this is actually used, it also does the position wrong.
+                int ctr = 1;
+                int UID=parseInt(request.getParameter("UID"));
+                while (new TagButton(UID, ctr).exists()) {
+                    ctr++;
+                }
+                new TagButton(UID, ctr, "new", "description");
+                out.print(""+ctr);
+                return;
+            }
             if (request.getParameter("projectID") != null) {
-                    int projectID=parseInt(request.getParameter("projectID"));
-                        int ctr = 1;
-                        while (new TagButton(projectID, ctr, true).exists()) {
-                            ctr++;
-                        }
-                        TagButton thisTag=new TagButton(projectID, ctr, "new", false, "description");
-                        out.print(""+ctr);
-                        return;
-
-                    }
+                int projectID=parseInt(request.getParameter("projectID"));
+                position = TagButton.getMaxPosition(projectID) + 1;
```

Comment:
It's not perfect, but this is a much shinier turd.

Reply:
Yep.  Not replacing this module, so gotta work with the crap.